packages/flutter_chat/lib/src/screens/chat_detail_screen.dart

```diff
@@ -0,0 +1,661 @@
+  final Function(Uint8List image) onUploadImage;
+  final Function(String text) onMessageSubmit;
+  final Function(ChatModel chat) onReadChat;
+
+  @override
+  State<ChatDetailScreen> createState() => _ChatDetailScreenState();
+}
+
+class _ChatDetailScreenState extends State<ChatDetailScreen> {
+  late String chatTitle;
+
+  @override
+  void initState() {
+    if (widget.chat.isGroupChat) {
+      chatTitle = widget.chat.chatName ??
+          widget.chatOptions.translations.groupNameEmpty;
+    } else {
+      chatTitle = widget.chat.users
+              .firstWhere((element) => element.id != widget.userId)
+              .fullname ??
+          widget.chatOptions.translations.anonymousUser;
+    }
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    var theme = Theme.of(context);
+
+    return widget.chatOptions.builders.chatDetailScaffoldBuilder?.call(
+          _AppBar(
+            chatTitle: chatTitle,
+            chatOptions: widget.chatOptions,
+            onPressChatTitle: widget.onPressChatTitle,
+            chatModel: widget.chat,
+          ) as AppBar,
+          _Body(
+            chatService: widget.chatService,
+            options: widget.chatOptions,
+            chat: widget.chat,
+            currentUserId: widget.userId,
+            onPressUserProfile: widget.onPressUserProfile,
+            onUploadImage: widget.onUploadImage,
+            onMessageSubmit: widget.onMessageSubmit,
+            onReadChat: widget.onReadChat,
+          ),
+          theme.scaffoldBackgroundColor,
+        ) ??
+        Scaffold(
+          appBar: _AppBar(
+            chatTitle: chatTitle,
+            chatOptions: widget.chatOptions,
+            onPressChatTitle: widget.onPressChatTitle,
+            chatModel: widget.chat,
+          ),
+          body: _Body(
+            chatService: widget.chatService,
+            options: widget.chatOptions,
+            chat: widget.chat,
+            currentUserId: widget.userId,
+            onPressUserProfile: widget.onPressUserProfile,
+            onUploadImage: widget.onUploadImage,
+            onMessageSubmit: widget.onMessageSubmit,
+            onReadChat: widget.onReadChat,
+          ),
+        );
+  }
+}
+
+class _AppBar extends StatelessWidget implements PreferredSizeWidget {
+  const _AppBar({
+    required this.chatTitle,
+    required this.chatOptions,
+    required this.onPressChatTitle,
+    required this.chatModel,
+  });
+
+  final String chatTitle;
+  final ChatOptions chatOptions;
+  final Function(ChatModel) onPressChatTitle;
+  final ChatModel chatModel;
+
+  @override
+  Widget build(BuildContext context) {
+    var theme = Theme.of(context);
+
+    return AppBar(
+      iconTheme: theme.appBarTheme.iconTheme ??
+          const IconThemeData(color: Colors.white),
+      centerTitle: true,
+      leading: GestureDetector(
+        onTap: () {
+          Navigator.popUntil(context, (route) => route.isFirst);
```

Comment:
This is only allowed if there is a root navigator specific for this user story. Otherwise you might pop the routes that are outside the scope of this user story.